### ISSUE TITLE
IndexError while running genie

### ISSUE THREAD

**Issue by Daki-23** [arkane-systems/genie#243]
**Windows version (build number):** Version 21H2 (OS Build 19044.1586)

**Linux distribution:** Ubuntu

**Genie version:** 2.2

**Describe the bug**

I'm unable to run any genie commands. Running the genie command results in an index out of range error

```
$genie --help
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/lib/genie/genie/__main__.py", line 791, in <module>
  File "/usr/lib/genie/genie/__main__.py", line 765, in entrypoint
  File "/usr/lib/genie/genie/__main__.py", line 309, in prelaunch_checks
IndexError: list index out of range
```

**Confirm that you are running inside the bottle:**
The output of `genie -b`.

**To Reproduce**
1. Run `genie --help` (or any other genie command)

**Expected behavior**
Should return a description of genie along with the accepted inputs and flags

**I confirm that I have read the ENTIRE supplied readme file and checked for relevant information on the repository wiki before raising this issue, and that if the solution to this issue is found in either location, it will be closed without further comment:**

 - [x] Yes.

**Comment by cerebrate** [comment]
Thank you for your report.

Could you let me have:

- the contents of your /etc/fstab
- the results of running `mount`
- the results of `python -V`
- the results of `apt show python3-psutil`

?

Thanks.

**Comment by Daki-23** [comment]
```
$ cat /etc/fstab
LABEL=cloudimg-rootfs   /        ext4   defaults        0 0
```

```
$ mount
rootfs on / type wslfs (rw,noatime)
none on /dev type tmpfs (rw,noatime,mode=755)
sysfs on /sys type sysfs (rw,nosuid,nodev,noexec,noatime)
proc on /proc type proc (rw,nosuid,nodev,noexec,noatime)
devpts on /dev/pts type devpts (rw,nosuid,noexec,noatime,gid=5,mode=620)
none on /run type tmpfs (rw,nosuid,noexec,noatime,mode=755)
none on /run/lock type tmpfs (rw,nosuid,nodev,noexec,noatime)
none on /run/shm type tmpfs (rw,nosuid,nodev,noatime)
none on /run/user type tmpfs (rw,nosuid,nodev,noexec,noatime,mode=755)
binfmt_misc on /proc/sys/fs/binfmt_misc type binfmt_misc (rw,relatime)
tmpfs on /sys/fs/cgroup type tmpfs (rw,nosuid,nodev,noexec,relatime,mode=755)
cgroup on /sys/fs/cgroup/devices type cgroup (rw,nosuid,nodev,noexec,relatime,devices)
C:\ on /mnt/c type drvfs (rw,noatime,uid=0,gid=0,case=off)
```

```
$python -v

Command 'python' not found, did you mean:

  command 'python3' from deb python3
  command 'python' from deb python-is-python3

$python3 --version
Python 3.8.10
```

```
$apt show python3-psutil
Package: python3-psutil
Version: 5.5.1-1ubuntu4
Priority: optional
Section: python
Source: python-psutil
Origin: Ubuntu
Maintainer: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Original-Maintainer: Sandro Tosi <morph@debian.org>
Bugs: https://bugs.launchpad.net/ubuntu/+filebug
Installed-Size: 901 kB
Depends: libc6 (>= 2.13), python3 (<< 3.9), python3 (>= 3.8~), python3:any
Suggests: python-psutil-doc
Homepage: https://github.com/giampaolo/psutil
Task: xubuntu-desktop, ubuntustudio-photography, ubuntukylin-desktop, ubuntu-mate-core, ubuntu-mate-desktop, ubuntu-budgie-desktop
Download-Size: 141 kB
APT-Manual-Installed: no
APT-Sources: http://archive.ubuntu.com/ubuntu focal/main amd64 Packages
Description: module providing convenience functions for managing processes (Python3)
 psutil is a module providing an interface for retrieving information on
 running processes and system utilization (CPU, memory) in a portable way
 by using Python, implementing many functionalities offered by tools like
 ps, top and Windows task manager.
 .
 It currently supports Linux, OS X, FreeBSD and Windows.
 .
 This package contains the Python 3 version of psutil.
```

**Comment by cerebrate** [comment]
Thanks. This is looking to me like it might be a psutil version issue - could you open up python3, execute the following, and send me the result for comparison? Thanks.

```
import psutil
psutil.disk_partitions()
```

**Comment by Daki-23** [comment]
```
$ python3
Python 3.8.10 (default, Nov 26 2021, 20:14:08)
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import psutil
>>> psutil.disk_partitions()
[]
```

**Comment by cerebrate** [comment]
...well, _that_ should definitely not be happening. I suspect this is a bug in psutil I'll need to take to upstream, but let's see if we can nail it down a little further? Does the `/etc/mtab` file exist on your system (probably as a symlink to `../proc/self/mounts`), and if so, do its contents match up to the output from `mount`?

Also, if you repeat the above test but change the second line to `psutil.disk_partitions(all=True)` does it still return an empty list?

**Comment by Daki-23** [comment]
The contents of /etc/mtab look to be the same as the output from mount AFAICT

```
$ cat /etc/mtab
rootfs / wslfs rw,noatime 0 0
none /dev tmpfs rw,noatime,mode=755 0 0
sysfs /sys sysfs rw,nosuid,nodev,noexec,noatime 0 0
proc /proc proc rw,nosuid,nodev,noexec,noatime 0 0
devpts /dev/pts devpts rw,nosuid,noexec,noatime,gid=5,mode=620 0 0
none /run tmpfs rw,nosuid,noexec,noatime,mode=755 0 0
none /run/lock tmpfs rw,nosuid,nodev,noexec,noatime 0 0
none /run/shm tmpfs rw,nosuid,nodev,noatime 0 0
none /run/user tmpfs rw,nosuid,nodev,noexec,noatime,mode=755 0 0
binfmt_misc /proc/sys/fs/binfmt_misc binfmt_misc rw,relatime 0 0
tmpfs /sys/fs/cgroup tmpfs rw,nosuid,nodev,noexec,relatime,mode=755 0 0
cgroup /sys/fs/cgroup/devices cgroup rw,nosuid,nodev,noexec,relatime,devices 0 0
C:\134 /mnt/c drvfs rw,noatime,uid=1000,gid=1000,case=off 0 0
```

```
$ python3
Python 3.8.10 (default, Nov 26 2021, 20:14:08)
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import psutil
>>> psutil.disk_partitions(all=True)
[sdiskpart(device='rootfs', mountpoint='/', fstype='wslfs', opts='rw,noatime'), sdiskpart(device='', mountpoint='/dev', fstype='tmpfs', opts='rw,noatime,mode=755'), sdiskpart(device='sysfs', mountpoint='/sys', fstype='sysfs', opts='rw,nosuid,nodev,noexec,noatime'), sdiskpart(device='proc', mountpoint='/proc', fstype='proc', opts='rw,nosuid,nodev,noexec,noatime'), sdiskpart(device='devpts', mountpoint='/dev/pts', fstype='devpts', opts='rw,nosuid,noexec,noatime,gid=5,mode=620'), sdiskpart(device='', mountpoint='/run', fstype='tmpfs', opts='rw,nosuid,noexec,noatime,mode=755'), sdiskpart(device='', mountpoint='/run/lock', fstype='tmpfs', opts='rw,nosuid,nodev,noexec,noatime'), sdiskpart(device='', mountpoint='/run/shm', fstype='tmpfs', opts='rw,nosuid,nodev,noatime'), sdiskpart(device='', mountpoint='/run/user', fstype='tmpfs', opts='rw,nosuid,nodev,noexec,noatime,mode=755'), sdiskpart(device='binfmt_misc', mountpoint='/proc/sys/fs/binfmt_misc', fstype='binfmt_misc', opts='rw,relatime'), sdiskpart(device='tmpfs', mountpoint='/sys/fs/cgroup', fstype='tmpfs', opts='rw,nosuid,nodev,noexec,relatime,mode=755'), sdiskpart(device='cgroup', mountpoint='/sys/fs/cgroup/devices', fstype='cgroup', opts='rw,nosuid,nodev,noexec,relatime,devices'), sdiskpart(device='C:\\', mountpoint='/mnt/c', fstype='drvfs', opts='rw,noatime,uid=1000,gid=1000,case=off')]
```

**Comment by cerebrate** [comment]
Well, I have good news and bad news.

The good news is that I've fixed the bug for 2.3; it turns out that filtering out non-hardware backed partitions, the way that _psutil_ does without that _all=True_ parameter, includes filtering out wslfs mounts.

The bad news is that your root partition is a wslfs mount, which means that you're running WSL 1. genie only works on WSL 2 (since 1 does not support needed system calls for systemd to work), so even with this bug fixed, all you'd get is an error message telling you that you need WSL 2.

Sorry.

(If you can upgrade the distro in question to WSL 2, you'll find the your current version executes without a problem; the bug only occurs if you're on WSL 1.)
